src/components/map-marker/map-marker.tsx

```diff
@@ -62,16 +69,38 @@ export class MapMarker implements GxComponent {
    */
   @Event() gxMapMarkerDeleted: EventEmitter;
 
+  private setPopup() {
+    const popupHtml = this.element.querySelector("[class='popupHtml']");
+    if (popupHtml.innerHTML) {
```

Comment:
What are we checking with this condition?

Reply:
basically, we check if users/developers have introduced some content inside ```<gx-map-maker>``` . If there is  any content (html), It calls leaflet method to add that content into the popup of marker (marker created by leaflet)

Reply:
Please consider that accessing the `innerHtml` property comes at a cost. If your just checking if it has content, there are other cheaper alternatives, like `firstElementChild !== null`.

Reply:
done in 7dfb8c6